mso/resource_mso_schema_site_anp.go

```diff
@@ -131,14 +131,27 @@ func resourceMSOSchemaSiteAnpCreate(d *schema.ResourceData, m interface{}) error
 	anpRefMap["templateName"] = anp_template_name
 	anpRefMap["anpName"] = anpName
 
-	path := fmt.Sprintf("/sites/%s-%s/anps/-", siteId, templateName)
-	anpStruct := models.NewSchemaSiteAnp("add", path, anpRefMap)
+	versionInt, err := msoClient.CompareVersion("4.0.0.0")
+	if err != nil {
+		return err
+	}
 
-	_, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaId), anpStruct)
+	if versionInt != 1 {
+		path := fmt.Sprintf("/sites/%s-%s/anps/%s", siteId, templateName, anpName)
+		anpStruct := models.NewSchemaSiteAnp("replace", path, anpRefMap)
+		_, err = msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", schemaId), anpStruct)
+	}
+
+	if versionInt == 1 || err != nil {
```

Comment:
What is the reason for the `err != nil` check?

Reply:
If the replace PATCH operation fails, we try will still try and add PATCH. This should never occur for ndo4 versions, but added as an additional safeguard.